CRAN Solaris error 3/24

Email...

> Please see the problems shown on
> <https://cran.r-project.org/web/checks/check_results_edina.html>.

`pow()` requires `doubles` not `int`. This was taken care of in ecdmcore... Just not propagated out. 

Log results... 

> /inst/include/edina_meat.h:27:33: error: call of overloaded ‘pow(int, unsigned int)’ is ambiguous
         vv(k) = pow(2, K - k - 1);

> /inst/include/edina_meat.h:38:41: error: call of overloaded ‘pow(int, unsigned int)’ is ambiguous
         double twopow = pow(2, K - k - 1);

> /inst/include/edina_meat.h:141:29: error: call of overloaded ‘pow(int, int&)’ is ambiguous
     double nClass = pow(2, K);

> /inst/include/edina_meat.h:154:29: error: call of overloaded ‘pow(int, unsigned int&)’ is ambiguous
     double nClass = pow(2, K);

> /inst/include/edina_meat.h:174:35: error: call of overloaded ‘pow(int, unsigned int&)’ is ambiguous
     unsigned int nClass = pow(2, K);

> /inst/include/edina_meat.h:175:37: error: call of overloaded ‘pow(int, unsigned int)’ is ambiguous
     unsigned int nqjs = pow(2, K - 1);

> /inst/include/edina_meat.h:258:29: error: call of overloaded ‘pow(int, unsigned int&)’ is ambiguous
     double nClass = pow(2, K);

> /inst/include/edina_meat.h:620:35: error: call of overloaded ‘pow(int, unsigned int&)’ is ambiguous
     unsigned int nClass = pow(2, K);

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -1,6 +1,6 @@
 Package: edina
 Title: Bayesian Estimation of an Exploratory Deterministic Input, Noisy and Gate Model
-Version: 0.1.0
+Version: 0.1.1
 Authors@R: c(
     person("James Joseph", "Balamuta", 
            email = "balamut2@illinois.edu", 

--- a/NEWS.md
+++ b/NEWS.md
@@ -4,6 +4,10 @@
 
 - Fix ambiguous overloaded of `pow` on Solaris.
 
+## Documentation
+
+- Added CRAN badges to README.
+
 # edina 0.1.0
 
 ## Features

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,3 +1,9 @@
+# edina 0.1.1
+
+## Bug Fix
+
+- Fix ambiguous overloaded of `pow` on Solaris.
+
 # edina 0.1.0
 
 ## Features

--- a/README.Rmd
+++ b/README.Rmd
@@ -17,6 +17,10 @@ knitr::opts_chunk$set(
 <!-- badges: start -->
 [![R build status](https://github.com/tmsalab/edina/workflows/R-CMD-check/badge.svg)](https://github.com/tmsalab/edina/actions)
 [![Package-License](http://img.shields.io/badge/license-GPL%20(%3E=2)-brightgreen.svg?style=flat)](http://www.gnu.org/licenses/gpl-2.0.html)
+[![CRAN Version Badge](http://www.r-pkg.org/badges/version/edina)](https://cran.r-project.org/package=edina)
+[![CRAN Status](https://cranchecks.info/badges/worst/edina)](https://cran.r-project.org/web/checks/check_results_edina.html)
+[![RStudio CRAN Mirror's Monthly Downloads](http://cranlogs.r-pkg.org/badges/edina?color=brightgreen)](http://www.r-pkg.org/pkg/edina)
+[![RStudio CRAN Mirror's Total Downloads](http://cranlogs.r-pkg.org/badges/grand-total/edina?color=brightgreen)](http://www.r-pkg.org/pkg/edina)
 <!-- badges: end -->
 
 Perform a Bayesian estimation of the Exploratory 

--- a/README.md
+++ b/README.md
@@ -8,6 +8,14 @@
 [![R build
 status](https://github.com/tmsalab/edina/workflows/R-CMD-check/badge.svg)](https://github.com/tmsalab/edina/actions)
 [![Package-License](http://img.shields.io/badge/license-GPL%20\(%3E=2\)-brightgreen.svg?style=flat)](http://www.gnu.org/licenses/gpl-2.0.html)
+[![CRAN Version
+Badge](http://www.r-pkg.org/badges/version/edina)](https://cran.r-project.org/package=edina)
+[![CRAN
+Status](https://cranchecks.info/badges/worst/edina)](https://cran.r-project.org/web/checks/check_results_edina.html)
+[![RStudio CRAN Mirror’s Monthly
+Downloads](http://cranlogs.r-pkg.org/badges/edina?color=brightgreen)](http://www.r-pkg.org/pkg/edina)
+[![RStudio CRAN Mirror’s Total
+Downloads](http://cranlogs.r-pkg.org/badges/grand-total/edina?color=brightgreen)](http://www.r-pkg.org/pkg/edina)
 <!-- badges: end -->
 
 Perform a Bayesian estimation of the Exploratory Deterministic Input,

--- a/cran-comments.md
+++ b/cran-comments.md
@@ -8,22 +8,8 @@
 
 0 errors | 0 warnings | 1 note
 
-- This is a new release and a resubmit to address CRAN remarks.
+> Days since last update: 0
 
-**Remarks**
-
-- The following misspelled words are due to the shortening of the list of authors
-  to "et al." and the method acronym.
-
-> Possibly mis-spelled words in DESCRIPTION:
-> EDINA (20:42)
-> al (21:53)
-> et (21:50)
-
-- From the last submission:
-
-- We've removed the `donttest` portion for `edina()` by dropping the number of
-  observations calculated to 1. We cannot remove the donttest option for 
-  `auto_edina()` as it requires at least 10 seconds with 2 different matrices
-  being estimated. 
-
+- This release addresses an issue uncovered with `pow()` on Solaris causing 
+  the package to fail to build.
+  https://www.r-project.org/nosvn/R.check/r-patched-solaris-x86/edina-00install.html

--- a/inst/include/edina_meat.h
+++ b/inst/include/edina_meat.h
@@ -24,7 +24,7 @@ inline arma::vec bijectionvector(unsigned int K)
     arma::vec vv(K);
 
     for (unsigned int k = 0; k < K; ++k) {
-        vv(k) = pow(2, K - k - 1);
+        vv(k) = std::pow(2.0, static_cast<double>(K - k) - 1.0);
     }
 
     return vv;
@@ -35,7 +35,7 @@ inline arma::vec inv_bijectionvector(unsigned int K, double CL)
     arma::vec alpha(K);
 
     for (unsigned int k = 0; k < K; ++k) {
-        double twopow = pow(2, K - k - 1);
+        double twopow = std::pow(2.0, static_cast<double>(K - k) - 1.0);
         alpha(k) = (twopow <= CL);
         CL = CL - twopow * alpha(k);
     }
@@ -138,7 +138,9 @@ inline arma::mat alpha_matrix(const arma::mat& Q) {
 
     int K = Q.n_cols;
 
-    double nClass = pow(2, K);
+    unsigned int nClass =
+        static_cast<unsigned int>(std::pow(2.0, static_cast<double>(K)));
+
     arma::mat alpha_mat(nClass, K);
 
     for(unsigned int cc = 0; cc < nClass; cc++){
@@ -151,7 +153,8 @@ inline arma::mat alpha_matrix(const arma::mat& Q) {
 
 inline arma::mat ETAmat(unsigned int K, unsigned int J, const arma::mat &Q)
 {
-    double nClass = pow(2, K);
+    unsigned int nClass =
+        static_cast<unsigned int>(std::pow(2.0, static_cast<double>(K)));
 
     arma::mat ETA(J, nClass);
 
@@ -171,8 +174,12 @@ inline arma::mat ETAmat(unsigned int K, unsigned int J, const arma::mat &Q)
 
 inline arma::cube ETAmat_nok(unsigned int K)
 {
-    unsigned int nClass = pow(2, K);
-    unsigned int nqjs = pow(2, K - 1);
+    unsigned int nClass =
+        static_cast<unsigned int>(std::pow(2.0, static_cast<double>(K)));
+
+    unsigned int nqjs =
+        static_cast<unsigned int>(std::pow(2.0, static_cast<double>(K) - 1.0));
+
     arma::vec zero_to_Km1 = arma::linspace(0, K - 1, K);
     arma::cube ETA_nok(K, nqjs, nClass);
     for (unsigned int k = 0; k < K; ++k) {
@@ -194,8 +201,12 @@ inline arma::cube ETAmat_nok(unsigned int K)
 
 inline arma::cube ETAmat_nok_one_m_ac(unsigned int K)
 {
-    unsigned int nClass = pow(2, K);
-    unsigned int nqjs = pow(2, K - 1);
+    unsigned int nClass =
+        static_cast<unsigned int>(std::pow(2.0, static_cast<double>(K)));
+
+    unsigned int nqjs =
+        static_cast<unsigned int>(std::pow(2.0, static_cast<double>(K) - 1.0));
+
     arma::vec zero_to_Km1 = arma::linspace(0, K - 1, K);
     arma::cube ETA_nok(K, nqjs, nClass);
     for (unsigned int k = 0; k < K; ++k) {
@@ -255,7 +266,9 @@ inline arma::vec abcounts(unsigned int N, const arma::vec &Yj, const arma::vec &
 
 inline arma::mat ClassbyQmat(unsigned int K)
 {
-    double nClass = pow(2, K);
+    unsigned int nClass =
+        static_cast<unsigned int>(std::pow(2.0, static_cast<double>(K)));
+
     arma::mat ETAbyQ(nClass, nClass - 1);
     for (unsigned int cc = 0; cc < nClass; ++cc) {
         arma::vec alpha_c = inv_bijectionvector(K, cc);
@@ -617,7 +630,8 @@ inline Rcpp::List edina_Gibbs_Q(const arma::mat &Y, unsigned int K,
     // unsigned int K = Q.n_cols;
 
     // Number of Classes
-    unsigned int nClass = pow(2, K);
+    unsigned int nClass =
+        static_cast<unsigned int>(std::pow(2.0, static_cast<double>(K)));
 
     // Total Chain length with burn
     // Prevents overflow

